src/lib/reporter.test.ts

```diff
@@ -0,0 +1,147 @@
+    },
+  },
+  providers: {
+    testprovider: {},
+  },
+});
+
+const mockProfileDocument: ProfileDocumentNode = {
+  kind: 'ProfileDocument',
+  header: {
+    kind: 'ProfileHeader',
+    scope: 'test',
+    name: 'profile',
+    version: {
+      major: 1,
+      minor: 0,
+      patch: 0,
+    },
+  },
+  definitions: [
+    {
+      kind: 'UseCaseDefinition',
+      useCaseName: 'Test',
+      safety: 'safe',
+      result: {
+        kind: 'UseCaseSlotDefinition',
+        type: {
+          kind: 'ObjectDefinition',
+          fields: [
+            {
+              kind: 'FieldDefinition',
+              fieldName: 'message',
+              required: true,
+              type: {
+                kind: 'NonNullDefinition',
+                type: {
+                  kind: 'PrimitiveTypeName',
+                  name: 'string',
+                },
+              },
+            },
+          ],
+        },
+      },
+    },
+  ],
+};
+
+const mockMapDocument: MapDocumentNode = {
+  kind: 'MapDocument',
+  header: {
+    kind: 'MapHeader',
+    profile: {
+      scope: 'test',
+      name: 'profile',
+      version: {
+        major: 1,
+        minor: 2,
+        patch: 3,
+      },
+    },
+    provider: 'test',
+  },
+  definitions: [
+    {
+      kind: 'MapDefinition',
+      name: 'Test',
+      usecaseName: 'Test',
+      statements: [
+        {
+          kind: 'OutcomeStatement',
+          isError: false,
+          terminateFlow: false,
+          value: {
+            kind: 'ObjectLiteral',
+            fields: [
+              {
+                kind: 'Assignment',
+                key: ['message'],
+                value: {
+                  kind: 'PrimitiveLiteral',
+                  value: 'hello',
+                },
+              },
+            ],
+          },
+        },
+      ],
+    },
+  ],
+};
+
+const mockServer = getLocal();
+
+describe('MetricReporter', () => {
+  it('should report metrics automagically', async () => {
+    SuperJson.loadSync = () => ok(mockSuperJson);
+    jest.mock('../client/registry', () => ({
+      ...jest.requireActual('../client/registry'),
+      getDefaultRegistryUrl: () => mockServer.url,
+    }));
+    await mockServer
+      .post('/sdk-events')
+      .always()
+      .thenCallback(async () => {
+        return {
+          status: 200,
+          body: 'hello',
+        };
+      });
+
+    const client = new SuperfaceClient();
+    (client as any).boundCache[
+      '{"id":"test-profile","version":"1.0.0"}{"name":"testprovider","security":[]}'
+    ] = new BoundProfileProvider(
+      mockProfileDocument,
+      mockMapDocument,
+      'testprovider',
+      {
+        security: [],
+      },
+      client
+    );
+    const profile = await client.getProfile('test-profile');
+    await profile.getUseCase('Test').perform();
+    await profile.getUseCase('Test').perform();
+    expect(client).not.toBe(undefined);
+    await new Promise(resolve => setTimeout(resolve, 20000));
```

Comment:
What we are testing here?

Reply:
nothing, there are no tests yet

Reply:
Well, that was funny to think about what do you want to do there😅